### ISSUE TITLE
Can't get solution to build.

### ISSUE THREAD

**Issue by manzanotti** [nachmore/unBand#68]
I've forked the code,installed the desktop app, installed the Microsoft.Band NuGet package, but the solution fails to build.

I can update the Microsoft.Band.* references for the unBand project to point to the ones in the desktop app folder, I get  a load of build errors:

'Microsoft.Band.Admin.CargoClient' is inaccessible due to its protection level	C:\Developement\unBand\src\unBand\BandHelpers\BandSensors.cs
 
Any suggestions as to what I've done wrong here?

**Comment by manzanotti** [comment]
NuGet package is version Microsoft.Band 1.3.10219-preview

The version of the assemblies in the desktop app folder are 1.3.10325.1, so is it the case that they've removed access to that class?

**Comment by robsonj** [comment]
Builds for me now, see pending pull req with my changes...

https://github.com/nachmore/unBand/pull/73

**Comment by manzanotti** [comment]
I've got 1.3.10423.3 of the Microsoft Band Sync app installed (the latest, as far as I can tell), and if I update all the Microsoft.Band.* references to point to the dlls I copied out of the Sync app folder, all I get are the same "X is inaccessible due to its protection level" errors. I've also updated all the Strapp* bits in the code that's changed to Tile*.

What am I missing here? If I look at the Dlls in ILSpy, then all the objects that VS is complaining about being internal (CargoClient, IDeviceTransport amongst others), and therefore inaccessible, are indeed internal, so I really shouldn't be able to see them!

**Comment by manzanotti** [comment]
Ok, I've changed all instances of CargoClient with ICargoClient, so those lines are now alright with the compiler (though I don't know if returning an internal class to something that is expecting a public interface will work at runtime), so I'm down to seven errors now:

Microsoft.Band.IDeviceTransport
Microsoft.Band.CargoStreamReader
Microsoft.Band.CargoStreamWriter
Microsoft.Band..CargoCommand

As these are all in the CargoClientExtender project, which doesn't appear to be used by the main project, I've just unloaded that project, and I can now build!

**Comment by robsonj** [comment]
There are 2 pending pull requests for build fixed and more build fixes, they address the initial build issues and the update to the latest dll's, or take a look at my branch for now. Until the pull requests get merged.. 

https://github.com/robsonj/unBand

**Comment by manzanotti** [comment]
Ah, I neglected to mention that I have looked at your two pull requests, and made the same changes, doesn't change the internal classes issues.

**Comment by robsonj** [comment]
You sure you got everything as it compiles fine for me on the latest ms dll's.

Try cloning my master branch.

**Comment by robsonj** [comment]
Just double checked, my master compiles fine for me on v1.3.10423.3

**Comment by manzanotti** [comment]
I created a branch locally from the current official master, pulled all your changes down, and I get exactly the same problems, which I can't say I'm that surprised about, given that looking at the DLLs in ILSpy matches up with the build errors I'm getting. The only thing I can think of is did you delete the bin and obj folders, maybe the earlier version DLL was cached? I've been deleting them, and it's not made a difference.

I've also updated Visual Studio to Update 4. This gave me the additional error of:

"SignFile" task was not given a value for the required parameter "CertificateThumbprint".

which I guess is the signing issue you mentioned in the other issue?

I'm officially flummoxed!

**Comment by robsonj** [comment]
The signfile error is just due to signing being turned on in the unband project. Just uncheck signing, but don't commit that setting.

When I get home I can zip up my working source and put it on Dropbox for you to look at. It's definitely using the same dll versions.

**Comment by manzanotti** [comment]
If you could, would really like to get to the bottom of this!

**Comment by robsonj** [comment]
Here you go...

https://www.dropbox.com/s/de6nw5nirk6jn4m/unband.zip?dl=0

**Comment by manzanotti** [comment]
Yep, I've downloaded your zip, and it builds fine! I'd love to know what the difference between mine and yours is, it really doesn't make any sense to me!

**Comment by manzanotti** [comment]
The Band DLLs I have, even though they are the same versions, are different, I have loads of classes and structs that are internal.

**Comment by robsonj** [comment]
Very odd. Have you tried a reinstall of the MS Band Sync software?
